cocotb/utils.py

```diff
@@ -498,6 +498,8 @@ def want_color_output():
     Colored output can be explicitly requested by setting :envvar:`COCOTB_ANSI_OUTPUT` to  ``1``.
     """
     want_color = sys.stdout.isatty()  # default to color for TTYs
+    if os.getenv("NO_COLOR") is not None:
+        want_color = False
```

Comment:
You probably want to put this between `COCOTB_ANSI_OUTPUT` and `GUI` so `NO_COLOR` has higher priority.

Reply:
We discussed this above, the spec says this position is correct, and that it should be deliberately lower priority